bokeh/plotting/figure.py

```diff
@@ -194,953 +195,18 @@ def __init__(self, *arg, **kw):
-
-        plot = figure(width=300, height=300)
-        plot.x(x=[1, 2, 3], y=[1, 2, 3], size=[10, 20, 25], color="#fa9fb5")
-
-        show(plot)
-
-"""
-
-    @marker_method()
-    def y(self, **kwargs):
-        """
-Examples:
-
-    .. bokeh-plot::
-        :source-position: above
-
-        from bokeh.plotting import figure, output_file, show
-
-        plot = figure(width=300, height=300)
-        plot.y(x=[1, 2, 3], y=[1, 2, 3], size=20, color="#DE2D26")
-
-        show(plot)
-
-"""
-
-    # -------------------------------------------------------------------------
-
-    @glyph_method(glyphs.Scatter)
-    def _scatter(self, **kwargs):
-        pass
-
-    def scatter(self, *args, **kwargs):
-        ''' Creates a scatter plot of the given x and y items.
-
-        Args:
-            x (str or seq[float]) : values or field names of center x coordinates
-
-            y (str or seq[float]) : values or field names of center y coordinates
-
-            size (str or list[float]) : values or field names of sizes in screen units
-
-            marker (str, or list[str]): values or field names of marker types
-
-            color (color value, optional): shorthand to set both fill and line color
-
-            source (:class:`~bokeh.models.sources.ColumnDataSource`) : a user-supplied data source.
-                An attempt will be made to convert the object to :class:`~bokeh.models.sources.ColumnDataSource`
-                if needed. If none is supplied, one is created for the user automatically.
-
-            **kwargs: :ref:`userguide_styling_line_properties` and :ref:`userguide_styling_fill_properties`
-
-        Examples:
-
-            >>> p.scatter([1,2,3],[4,5,6], marker="square", fill_color="red")
-            >>> p.scatter("data1", "data2", marker="mtype", source=data_source, ...)
-
-        .. note::
-            When passing ``marker="circle"`` it is also possible to supply a
-            ``radius`` value in data-space units. When configuring marker type
-            from a data source column, *all* markers including circles may only
-            be configured with ``size`` in screen units.
-
-        '''
-        marker_type = kwargs.pop("marker", "circle")
-
-        if isinstance(marker_type, str) and marker_type in _MARKER_SHORTCUTS:
-            marker_type = _MARKER_SHORTCUTS[marker_type]
-
-        # The original scatter implementation allowed circle scatters to set a
-        # radius. We will leave this here for compatibility but note that it
-        # only works when the marker type is "circle" (and not referencing a
-        # data source column). Consider deprecating in the future.
-        if marker_type == "circle" and "radius" in kwargs:
-            return self.circle(*args, **kwargs)
-        else:
-            return self._scatter(*args, marker=marker_type, **kwargs)
+    def mapping(self, *, x_range: Range, y_range: Range, x_target: Range, y_target: Range) -> GlyphAPI:
```

Comment:
I changed this for now, but I will revisit this with the next round of sub-coordinates' related additions, because `subplot` may have a better usage in other context.